src/main/java/edu/harvard/iq/dataverse/api/Datasets.java

```diff
@@ -2502,6 +2504,67 @@ public Response addFileToDataset(@PathParam("id") String idSupplied,
     } // end: addFileToDataset
 
 
+    /**
+     * Clean storage of a Dataset
+     *
+     * @param idSupplied
+     * @return
+     */
+    @GET
+    @Path("{id}/cleanStorage")
+    public Response cleanStorage(@PathParam("id") String idSupplied, @QueryParam("dryrun") Boolean dryrun) {
+        // get user and dataset
+        User authUser;
+        try {
+            authUser = findUserOrDie();
+        } catch (WrappedResponse ex) {
+            return error(Response.Status.FORBIDDEN,
+                    BundleUtil.getStringFromBundle("file.addreplace.error.auth")
+            );
+        }
+
+        Dataset dataset;
+        try {
+            dataset = findDatasetOrDie(idSupplied);
+        } catch (WrappedResponse wr) {
+            return wr.getResponse();
+        }
+        
+        // check permissions
+        if (!permissionSvc.permissionsFor(createDataverseRequest(authUser), dataset).contains(Permission.EditDataset)) {
+            return error(Response.Status.INTERNAL_SERVER_ERROR, "Access denied!");
+        }
+
+        List<String> deleted = new ArrayList<>();
+        Set<String> files = new HashSet<String>();
+        try {
+            for (DatasetVersion dv : dataset.getVersions()) {
```

Comment:
Is there a reason for iterating through the versions->FileMetadatas here? 
As opposed to just 
```
for (DataFile dataFile : dataset.getFiles()) {
```
? - then you wouldn't have to iterate through the same files that occur in more than one versions. Not a big deal, obviously.

Reply:
I did not know I could do that, I have implemented your suggestion. Thank you.